### DISCUSSION TITLE
Вопросы программирования

### DISCUSSION THREAD

**Discussion by gtfox** [zamtmn/zcad#130] [Q&A] [answered]
Команды не отображаются в инспекторе.
Интересно, так и должно быть?
![Безымянный](https://github.com/user-attachments/assets/19f1a334-dfdf-4eea-b902-0cc8e1650ace)

**Comment by zamtmn** [comment, marked as answer]
Да, так и должно быть. Инспектор проекта отображает не всё что uses, а всё что добавлено в проект
![image](https://github.com/user-attachments/assets/edf5eb33-affc-4212-aee6-8321721521cf)
т.е. открыть соответствующий файл в редакторе, нажать щыфт+ф11, сохранить изменения в файле проекта и закомитить.

Добавление в проект нужно для возможности навигации по этому дереву и для работы системы локализации программы, т.е. все файлы с формами надо добавлять в проект чтоб их можно было перевести. остальные файлы - по желанию.

возможно я не знаю еще какие-нибудь плюшки от добавления в проект и не пользую особо это

**Comment by gtfox** [comment]
Я тут долго пытался найти как покрасить зарезервированные слова в разные цвета (потому, что в тырнете видел на картинках)
И тут случайно нашел ответ :) https://t.me/zcad_dev/524
Вот эта настройка включает: Отображать контур блока

![01](https://github.com/user-attachments/assets/b16bbdef-cfc8-46f1-b453-5ba71d447120)

Но плюсом к полоске она еще красит код как новогоднюю ёлку :)
![02](https://github.com/user-attachments/assets/3a8cf4f1-4b48-4e54-8021-ea2d4fe7a271)
Ну и сама тема от Андрея чуть изменённая мной
[forMDS_FOX.zip](https://github.com/user-attachments/files/17356148/forMDS_FOX.zip)

**Comment by zamtmn** [comment]
Про кнопку контуров я всё время забываю и приходится ее каждый раз искать. надо было спросить))
По цветовой теме - у меня стандартная лазарьная темная, толко цвет фона на ней изменен чтоб не отличаться от мдс. коментарии в неактивном deffined блоке в ней сливаются с фоном))

**Reply by gtfox** [reply]
> надо было спросить))

если постоянно всё спрашивать можно моск за..рахать :)

> коментарии в неактивном deffined блоке в ней сливаются с фоном))

это мы переживём :)
Но ёлка мне зашла, аж настроение улучшилось :)

**Comment by gtfox** [comment]
Есть DBGrids.pas, который реализует отображение Dataset-ов в виде таблицы. 
Но в нём нет автоподбора ширины столбцов по содержимому. Вернее есть, но он не доступен извне. Можно только сделать даблклик на разделителе заголовков столбцов. 
Люди в тырнетах сами пишут автоподборы.
Один из них обратился к автору и попросил реализовать это в DBGrids.pas
https://gitlab.com/freepascal.org/lazarus/lazarus/-/issues/30061
В конце обсуждения выложен патч.

Файл в лазарусе 2003
Обсуждалось в 2016
Выложили в 2021

И тут есть куча вопросов:
1. Как применить этот патч? Вручную конечно можно перекопировать код, но интересно как это задумывалось.
2. Он изменит локальный файл? И при переустановке лазаруса его надо сохранять и возвращать? На странице есть версия 1.8, но в лазарусе в файле нет упоминания версий.
4. Почему в лазарусе нет последней версии или автор вообще не коммитил эти изменения?

Я воспользовался одним из вариантов из тырнета, но просто интересно, может я чего-то недопонял...

**Reply by Alligator-1** [reply]
Хм...
![изображение](https://github.com/user-attachments/assets/7280c26e-afc1-4e21-827f-70cdc675f503)

**Reply by gtfox** [reply]
Блин, так и есть. Я зарылся в поиски и не видел ответа на поверхности. Я искал Auto Size Columns и прочие вариации.
Так еще и на паре десятков страниц люди, в основном писали свои костыли, и это наводило на мысль, что так и надо :)
Интересно, что вчера попытки заставить работать свойство dgAutoSizeColumns не приводили к результату, т.к. в примере, который я ковырял, был код DBGrid.Columns[i].Width := 100;

И самое смешное, что правильный ответ AutoAdjustColumns всплывал всего на паре страниц в тырнете, но на стековерфлоу его не то, что пометить, заминусовать умудрились :)
![01](https://github.com/user-attachments/assets/2aa43cb7-e73c-4844-92d5-2c57c675660e)

Спасибо за помощь!

**Comment by zamtmn** [comment]
1 - есть много способов. самый простой смотри в контекстном меню черепахогита. но т.к. паич старый, не фкт что он наложится на без проблем и конфликтов
2 - да, он изменит твои файлы. после переустановки лазаря патч надо будет наложить заново
3 - хз, я этот компонент не юзаю не видел, истории его не знаю

**Reply by gtfox** [reply]
Точно, нашел :)
![01](https://github.com/user-attachments/assets/272abfea-6944-4050-a7b1-11347d14a7b0)
Понял, спасибо! Вопрос решился выше :)
